test/e2e/network/ingress.go

```diff
@@ -1044,3 +1118,38 @@ func detectHttpVersionAndSchemeTest(f *framework.Framework, jig *ingress.Ingress
 	})
 	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to get %s or %s, response body: %s", version, scheme, resp))
 }
+
+func detectNegAnnotation(f *framework.Framework, jig *ingress.IngressTestJig, gceController *gce.GCEIngressController, ns, name string, negs int) {
+	wait.Poll(5*time.Second, framework.LoadBalancerPollTimeout, func() (bool, error) {
+		svc, err := f.ClientSet.CoreV1().Services(ns).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		// if we expect no NEGs, then we should be using IGs
+		if negs == 0 {
+			return gceController.BackendServiceUsingIG(jig.GetServicePorts(false))
+		}
+
+		var status ingress.NegStatus
+		v, ok := svc.Annotations[ingress.NEGStatusAnnotation]
+		if !ok {
+			framework.Logf("Waiting for %v, got: %+v", ingress.NEGStatusAnnotation, svc.Annotations)
+			return false, nil
+		}
+
+		err = json.Unmarshal([]byte(v), &status)
+		if err != nil {
+			framework.Logf("Error in parsing Expose NEG annotation: %v", err)
+			return false, nil
+		}
+		framework.Logf("Got %v: %v", ingress.NEGStatusAnnotation, v)
+
+		if len(status.NetworkEndpointGroups) != negs {
```

Comment:
Besides comparing the annotation from signal. It is possible to check if the NEG resources exists.